### ISSUE TITLE
Linux Mint support

### ISSUE THREAD

**Issue by maroneze** [ocaml/opam-depext#26]
Is Linux Mint explicitly supported or unsupported by depext?

I'm asking because I had an issue with an unlisted dependence on a package, but I don't know if it's the package manager who should add "linuxmint" to his package dependencies, or if a patch could be applied to depext itself (e.g. to recognize it as "ubuntu", or something of the sort).

If so, I can try to propose a patch for that. But if it's not worth supporting yet another distro, it's ok for me.

**Comment by avsm** [comment]
Not supported right now, but patches are most welcome.  Is there a Docker image for Linux Mint, so it can be rotated into the testing pool?

**Comment by AltGr** [comment]
Definitely worth it, thanks for posting!

**Comment by ghost** [comment]
I didn't find any Docker images for Linux Mint.
Concerning the patch, I think it should be fairly safe to consider Linux Mint as Ubuntu for most practical purposes.

**Comment by AltGr** [comment]
In this case, we may want to just tweak opam-depext to consider Mint as Ubuntu rather than updating all packages' depexts; we could still refine afterwards if they were to diverge.

**Comment by ghost** [comment]
I'm not following exactly what the code does (I'm reading the code on Github, but I have depext 0.5 installed), but it already seems able to detect it as Ubuntu (`distribution` returns `Ubuntu`), although when I run `opam depext -d` it displays:

```
+ uname -m
+ uname -s
+ lsb_release -i -s
# Detecting depexts using flags: x86_64 linux linuxmint
+ opam list --safe --recursive --external=x86_64,linux,linuxmint --required-by=
+ opam list --safe --recursive --external=source,x86_64,linux,linuxmint --required-by=
# No extra OS packages requirements found.
```

`lsb_release` does return `LinuxMint`, but `/etc/os-release` exists and it contains:

```
NAME="Ubuntu"
VERSION="14.04.1 LTS, Trusty Tahr"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 14.04.1 LTS"
VERSION_ID="14.04"
```

So I'm not sure why the code is not taking this into account and running `lsb_release` instead.

In particular, I tested it with package `curses`, which has `["ubuntu"] ["libncurses-dev"]` in its depexts, but it tells me "No extra OS packages requirements found.", even when that library is not installed.

**Comment by AltGr** [comment]
yes, this has been fixed since 0.5
pushing a new release just now :)

**Comment by ghost** [comment]
Excellent, just tested and works fine.

The only minor detail is that, because the exact package is called "libncurses5-dev" instead of "libncurses-dev" as defined in curses' depexts, running `opam depext curses` displays:

```
# The following system packages are needed:
#  - libncurses-dev
#  - m4
# The following new OS packages need to be installed: libncurses-dev
```

Apt-get then finds `libncurses5-dev` as synonym of `libncurses-dev` and reinstalls it.

But since in practice depext will only run if the OPAM package does not build successfully, this will hardly if ever be seen by anyone.

By the way, it seems that a "robust" solution would require using `aptitude` ([as indicated in this Stack Exchange question](http://unix.stackexchange.com/questions/96309/how-can-i-know-if-a-virtual-package-is-installed-on-a-debian-system) to query for `Provides` fields to solve this specific problem, but I don't think it's worth dragging another command and many more lines of code into depext for such a minor annoyance.
